smash/io/hdf5_io.py

```diff
@@ -0,0 +1,1041 @@
+    if typeofstructure=="light":
+        
+        key_data=generate_light_smash_object_structure(structure)
+        
+    elif typeofstructure=="medium":
+        
+        key_data=generate_medium_smash_object_structure(structure)
+        
+    elif typeofstructure=="full":
+        
+        key_data=generate_full_smash_object_structure(instance)
+    
+    return key_data
+
+
+
+
+def dump_object_to_hdf5_from_list_attribute(hdf5,instance,list_attr):
+    """
+    dump a object to a hdf5 file from a list of attributes
+
+    Parameters
+    ----------
+    hdf5 : object
+        an hdf5 object
+    instance : object
+        a custom python object.
+    list_attr : list
+        a list of attribute
+    """
+    if isinstance(list_attr,list):
+    
+        for attr in list_attr:
+            
+            if isinstance(attr, str):
+                
+                dump_object_to_hdf5_from_str_attribute(hdf5, instance, attr)
+            
+            elif isinstance(attr,list):
+                
+                dump_object_to_hdf5_from_list_attribute(hdf5, instance, attr)
+            
+            elif isinstance(attr,dict):
+                
+                dump_object_to_hdf5_from_dict_attribute(hdf5, instance, attr)
+                
+            else:
+                
+                raise ValueError(
+                    f"unconsistant {attr} in {list_attr}. {attr} must be a an instance of dict, list or str"
+                )
+    
+    else:
+        
+        raise ValueError(
+                    f"{list_attr} must be a instance of list."
+                )
+
+
+
+def dump_object_to_hdf5_from_dict_attribute(hdf5,instance,dict_attr):
+    """
+    dump a object to a hdf5 file from a dictionary of attributes
+
+    Parameters
+    ----------
+    hdf5 : object
+        an hdf5 object
+    instance : object
+        a custom python object.
+    dict_attr : dict
+        a dictionary of attribute
+    """
+    if isinstance(dict_attr,dict):
+    
+        for attr, value in dict_attr.items():
+            
+            hdf5=add_hdf5_sub_group(hdf5, subgroup=attr)
+            
+            try:
+            
+                sub_instance=getattr(instance, attr)
+                
+            except:
+                
+                sub_instance=instance
+            
+            if isinstance(value,dict):
+            
+                dump_object_to_hdf5_from_dict_attribute(hdf5[attr], sub_instance, value)
+            
+            if isinstance(value,list):
+            
+                dump_object_to_hdf5_from_list_attribute(hdf5[attr], sub_instance, value)
+            
+            elif isinstance(value,str):
+                
+                dump_object_to_hdf5_from_str_attribute(hdf5[attr], sub_instance, value)
+            
+            else :
+                
+                raise ValueError(
+                    f"Bad type of '{attr}' in '{dict_attr}'. Dict({attr}) must be a instance of dict, list or str"
```

Comment:
Juste pour être homogène avec la fonction **dump_object_to_hdf5_from_list_attribute**
```suggestion
                    f"inconsistent '{attr}' in '{dict_attr}'. Dict({attr}) must be a instance of dict, list or str"
```